File: MPI/c/hello2.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
  int ierr, num_procs, my_id;

  // Create child processes (# specified at runtime). From this point on,
  // every processes executes a separate copy of this program; no variables
  // are shared
  ierr = MPI_Init(&argc, &argv);

  // Find out process ID of current processor, and total # of processes
  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &my_id);
  ierr = MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
  
  printf("Hello world! I'm process %i out of %i processes\n",
          my_id, num_procs);

  if( my_id == 0 )
  {
    printf("\nI am process 0!\n");
  }
  
  // Stop this process
  ierr = MPI_Finalize();
}

File: MPI/fortran/hello2.f
c     Another basic 'Hello World' program

      program helloworld2
    
      include '/usr/lib/x86_64-linux-gnu/openmpi/include/mpif.h'

      integer ierr       ! error status for initializing MPI
      integer num_procs  ! number of processes used
      integer my_id      ! get ID/rank of the current process

      call MPI_INIT ( ierr )  ! initialize MPI

c     Get process ID of the current process
      call MPI_COMM_RANK (MPI_COMM_WORLD, my_id, ierr)

c     Get total number of processes
      call MPI_COMM_SIZE (MPI_COMM_WORLD, num_procs, ierr)

      print *, "Hello world! I'm process ", my_id, " out of ",
     &             num_procs, " processes."

      call MPI_FINALIZE ( ierr )  ! close MPI

      stop
      end
